### ISSUE TITLE
Cannot delete environment tag in UI when it is imported via PublicAPI

### ISSUE THREAD

**Issue by mkralik3** [syndesisio/syndesis#4735]
## This is a...

<pre><code>
[ ] Feature request
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Documentation issue or request
</code></pre>

## Description
When I created some environment tag (ENV1) in CI/CD dialog, i am able to delete it. 
However, when I import some integrations through PublicAPI and I specify another environment tag (ENV2), 
`
curl ... https://public-syndesis.my-minishift.syndesis.io/api/v1/public/integrations -F data=@export.zip -F environment=ENV2
`
I am not able to delete it (ENV2) via UI dialog even though the imported integrations are already deleted.
I am able to delete it only via PublicAPI
`
curl ... -X DELETE https://public-syndesis.my-minishift.syndesis.io/api/v1/public/environments/ENV2 
`
![output](https://user-images.githubusercontent.com/16251792/53741896-ade59980-3e97-11e9-906c-487fa3b820fc.gif)

**Comment by heiko-braun** [comment]
> I am not able to delete it (ENV2) via UI dialog even though the imported integrations are already deleted. I am able to delete it only via PublicAPI

@gashcrumb A UI issue maybe?


@mkralik3 Is there anything in the Browser JavaScript log?

**Comment by mkralik3** [comment]
yes,
```
Error performing DELETE request to https://syndesis.my-minishift.syndesis.io/api/v1/public/integrations/i-L_8h3seuUikdLP8f5WHz/tags/ENV2 :  {…}​error: Object { errorCode: 404, userMsg: "Given request is not acceptable", developerMsg: "Missing environment tag ENV2" }​headers: Object { normalizedNames: Map(0), lazyUpdate: null, lazyInit: lazyInit() }​message: "Http failure response for https://syndesis.my-minishift.syndesis.io/api/v1/public/integrations/i-L_8h3seuUikdLP8f5WHz/tags/ENV2: 404 Not Found"​name: "HttpErrorResponse"​ok: false​status: 404​statusText: "Not Found"​url: "https://syndesis.my-minishift.syndesis.io/api/v1/public/integrations/i-L_8h3seuUikdLP8f5WHz/tags/ENV2"​<prototype>: Object { constructor: t() } main.caa99a5dffebc3f3a9a7.js:1:2961612
```
```
{
  "headers": {
    "normalizedNames": {},
    "lazyUpdate": null
  },
  "status": 404,
  "statusText": "Not Found",
  "url": "https://syndesis.my-minishift.syndesis.io/api/v1/public/integrations/i-L_8h3seuUikdLP8f5WHz/tags/ENV2",
  "ok": false,
  "name": "HttpErrorResponse",
  "message": "Http failure response for https://syndesis.my-minishift.syndesis.io/api/v1/public/integrations/i-L_8h3seuUikdLP8f5WHz/tags/ENV2: 404 Not Found",
  "error": {
    "errorCode": 404,
    "userMsg": "Given request is not acceptable",
    "developerMsg": "Missing environment tag ENV2"
  }
}
```

It looks that UI dialog uses rest point which removes environment tag only on integration 
`DELETE ​/public​/integrations​/{id}​/tags​/{env}`
However, that tag is not defined on that integration ('c' in my example) so it returns 404. 

In my point of view, unchecking environment tag (which doesn't do anything this time https://github.com/syndesisio/syndesis/issues/4683) should call this endpoint 
`DELETE ​/public​/integrations​/{id}​/tags​/{env}`
because it removes the tag only from the integration.

And Remove button should call this endpoint
`DELETE /public​/environments​/{env}`
which remove environment tag from whole integration.

@dhirajsb described it here https://github.com/syndesisio/syndesis/issues/4683#issuecomment-468404868

**Comment by dhirajsb** [comment]
@mkralik3 it looks like a bug. The cache needs to be updated after importing.

**Comment by dhirajsb** [comment]
@mkralik3 and it also possibly needs to filter out deleted integrations when maintaining the cache.

**Comment by dhirajsb** [comment]
Added PR #4745 to remove environment name caching. Proposed fix to UI discussed in #4683 should also help resolve the inconsistent behavior it shows right now.

**Comment by mkralik3** [comment]
Verified. When I import the integration and after that, I delete it, the tags are deleted with it the integration. However, another bug was rising: https://github.com/syndesisio/syndesis/issues/4755
